Src/Particle/AMReX_NeighborList.H

```diff
@@ -10,77 +10,127 @@ namespace amrex
 
 template <class ParticleType>
 struct Neighbors
-{    
+{
     struct iterator
     {
         AMREX_GPU_HOST_DEVICE
-        iterator (int start, int stop, const unsigned int * nbor_list_ptr, const ParticleType* pstruct) 
-            : m_index(start), m_stop(stop), m_nbor_list_ptr(nbor_list_ptr), m_pstruct(pstruct)   
+        iterator (int start, int stop, const unsigned int * nbor_list_ptr, ParticleType* pstruct)
+            : m_index(start), m_stop(stop), m_nbor_list_ptr(nbor_list_ptr), m_pstruct(pstruct)
         {}
-        
+
         AMREX_GPU_HOST_DEVICE
         void operator++ () { ++m_index;; }
-        
+
         AMREX_GPU_HOST_DEVICE
         bool operator!= (iterator const& rhs) const { return m_index < m_stop; }
-        
+
         AMREX_GPU_HOST_DEVICE
-        ParticleType operator* () const { return m_pstruct[m_nbor_list_ptr[m_index]];  }
-        
+        const ParticleType& operator* () const { return m_pstruct[m_nbor_list_ptr[m_index]];  }
+
+        AMREX_GPU_HOST_DEVICE
+        ParticleType& operator* () { return m_pstruct[m_nbor_list_ptr[m_index]];  }
```

Comment:
I think there should be only one version `ParticleType& operator* () const`.  `Iterator const` should still return non-const reference.  User could assign to to `const&` if they want.

See https://en.cppreference.com/w/cpp/iterator/iterator

Reply:
Okay, done.